crates/next-dev/benches/util/mod.rs

```diff
@@ -131,105 +132,131 @@ pub fn resume_on_error<F: FnOnce() + UnwindSafe>(f: F) {
-                        retry_async_default((), |_| setup())
-                            .await
-                            .expect("failed to setup"),
-                    );
-                    let input = black_box(warmup(input).await).expect("failed to warmup");
-
-                    let start = early_start.unwrap_or_else(|| measurement.start());
-                    match routine(input).await {
-                        Ok(output) => {
-                            let duration;
-                            if bench_benchmark_itself {
-                                teardown(black_box(output)).await;
-                                duration = measurement.end(start);
-                            } else {
-                                duration = measurement.end(start);
-                                teardown(black_box(output)).await;
-                            }
-                            if log_progress {
-                                eprint!(" {} ", FormatDuration(duration));
-                            }
-                            value = measurement.add(&value, &duration);
-                            iter += 1;
-                            break;
-                        }
-                        Err(err) => {
-                            failures += 1;
-                            if failures > iters {
-                                panic!("Routine failed {failures} times, aborting\n{:?}", err)
-                            } else {
-                                eprintln!("Routine failed, will be retried: {:?}", err);
-                                continue;
-                            }
-                        }
+
+            loop {
+                let start = measurement.start();
+                let output = routine(input).await.expect("Routine failed");
+                let duration = measurement.end(start);
+
+                value = measurement.add(&value, &duration);
+                iter += 1;
+
+                if iter < iters {
+                    if log_progress_verbose && iter.count_ones() == 1 {
+                        eprint!(" [{}/{}]", FormatDuration(value / iter as u32), iter);
                     }
+
+                    input = restore(black_box(output)).await.expect("failed to restore");
+                } else {
+                    if log_progress {
+                        eprint!(" {}", FormatDuration(value / iter as u32),);
+                    }
+                    output_mutex.lock().unwrap().replace(output);
+                    break;
                 }
             }
 
             value
-        })
+        });
+
+        let measurement = WallTime;
+        let output = output_mutex
+            .lock()
+            .unwrap()
+            .take()
+            .expect("iter_custom must execute it's closure");
+        if log_progress_verbose {
+            eprint!(" teardown...");
+        }
+        let start = measurement.start();
```

Comment:
For these debug logs, we don't need to use `measurement`, we can use `Instant::now()/elapsed()`.